### PR TITLE
Rebuild for Python 3.11

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Simple rebuild for Python 3.11. I also adjusted the runtime dependencies to better match the upstream package.

### CODE REVIEW

**Review by ELundby45 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ requirements:
     - wheel
   run:
     - python
-    - numpy
+    - numpy >=1.17.3
     - scipy >=1.3.2
     - scikit-learn >=1.0.2
     - joblib >=1.1.1
```
Comment (ryanskeith) [inline recipe/meta.yaml:30]: May be missing threadpoolctl. https://github.com/scikit-learn-contrib/imbalanced-learn/blob/8a42dfc74d5b9408d2da922cd88f645c428c5d26/imblearn/_min_dependencies.py#L22

**Review by ryanskeith — COMMENTED** [review]
If I am reading this correctly, the recipe is missing threadpoolctl.

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:30]: Yeah, I did look at this. I think the reason it's missing is that it's not imported in the code and if I'm not mistaken, threadpoolctl isn't a "magic" library that works without importing it? Maybe we should just use a `run_constrained`?

Comment (skupr-anaconda) [reply recipe/meta.yaml:30]: It's needed only for Azure https://github.com/search?q=repo%3Ascikit-learn-contrib%2Fimbalanced-learn+threadpoolctl&type=code. 
I would put it to `run_constrained`

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**